### ISSUE TITLE
packing in ci

### ISSUE THREAD

**Issue by gedw99** [dawenxi-tech/2fa#24]
packaging  for Mac ( DMG ) , windows and linux

activated for Mac only as I only has a Mac, and it works on laptop and in ci.

Have not worked out out where I want to release to. For now I guess GitHub release tags.

@heramerom @DawenxiTech 

If you want to try locally  then turn it on in the make file, and if it works then it will work in ci.

**Comment by heramerom** [comment]
Thanks, that's works perfect on my laptop.

**Comment by gedw99** [comment]
thanks for letting me know it works on your laptop. Windows or Mac . amd64 right ?

If you can get it running in CI then ? Just edit the makefile and CI will do it.

The packing phase for DMG, etc is currently only turned on for GitHub tags btw can you can the Pack target locally first and then if it works turn it on for CI by tagging a release.

We need to still modify the GitHub ci to release artefacts btw..

**Comment by gedw99** [comment]
https://github.com/dawenxi-tech/2fa/blob/main/Makefile#L234C5-L234C5

and 

https://github.com/dawenxi-tech/2fa/blob/main/Makefile#L282

**Comment by heramerom** [comment]
I'm on macos amd64. Thank you.

I don't know how to do, I give you admin rights and you can push tags now.

**Comment by gedw99** [comment]
It’s already on in CI. So I don’t need Admin rights.

just modify the make file so that windows is included in the build and pack targets. You can see how its sniffs what OS is running and then calls the is specific make target for build and package.

Just follow my lead of how I setup the make file and go for it.

I think the way the make file works is a decent  technique. Mac does full build, package currently locally and in CI.  

release is not quite there yet :)  we get that working later ..

Can you try it and let me know.  I am pretty busy with travel / work stuff in the UK right now, but can help if you get stuck.

**Comment by heramerom** [comment]
Yeah, thank you! I'm tried and it works perfect on my macos amd64.

Have a nice trip!
